### DISCUSSION TITLE
Assign complex crystal orientations

### DISCUSSION THREAD

**Discussion by Ming-is** [neperfepx/neper#439] [Q&A] [answered]
The microstructure is represented by a 1000-grain 3-D Voronoi tessellation.
I want to assign custom crystal orientation to about 100 grains in the center and random orientation to the rest. The composition of custom orientation is:         "0.3*fiber(1,0,0,0,0,1)+0.2*fiber(1,1,0,0,0,1)+0.5*fiber(1,1,1,0,0,1)"

How can I achieve it?

![3 DYVRSS@49 5Z{``DJNB4H](https://user-images.githubusercontent.com/71123582/163186649-44b0b4b5-d9da-4e6e-b5b9-4fe164e4433e.png)

**Comment by rquey** [comment, marked as answer]
This requires a few steps:

1/ Generate the custom orientation distribution for whatever many grains you need.
2/ Generate a random orientation distribution for the rest.
3/ Generate the tessellation and output the grain properties, e.g. x, y, z, to identify which grains are central (to be assigned the custom orientations).
4/ Based on the grain properties, write an orientation file that contains the list of grain orientations.
5/ Generate again the tessellation and assign it the desired orientations using `-ori file()`.
